frontend/src/hooks/useNavigation.js

```diff
@@ -0,0 +1,21 @@
+import { useState } from "react";
+import { useHistory } from "react-router";
+
+const useNavigation = modalPath => {
```

Comment:
The logic of this hook is still divided unnecessarily in the component. To simplify things, this hook should return four things:

1. State of dialog - if it is open or not
2. Function to close the dialog
3. Function to navigate back directly (this should also always close the dialog)
4. Function to navigate conditionally, either navigate or open dialog (this can be merged into one function with 3.)

This should simplify this code a little and you can remove all the remaining logic from the component itself. Do not be afraid to repeat some lines of code to avoid convoluted logic like you have currently.

Reply:
You're on the right track, a few more improvements and this will be a nice hook.